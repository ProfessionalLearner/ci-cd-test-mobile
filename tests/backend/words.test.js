const request = require('supertest');
const app = require('../../backend/app');
const config = require('config');
const mongoose = require('mongoose');

describe("Test api/words/add route", () => {
    beforeAll( async () => {
        await mongoose.connect(config.get('testDB'), {
            useNewUrlParser: true,
            useUnifiedTopology: true,
            useFindAndModify: false,
        });
    })


    async function removeAllCollections () {
        const collections = Object.keys(mongoose.connection.collections);
        for (const collectionName of collections) {
          const collection = mongoose.connection.collections[collectionName];
          await collection.deleteMany();
        }
    }

    afterAll(async () => {
        await removeAllCollections();
        await mongoose.connection.close();
    })

    test("should be invalid input", async () => {
        const res = await request(app)
                    .post('/api/words/add')
                    .send({
                        description: "@cas"
                    })
        expect(res.status).toBe(400);
    })

    test("should add new word", async() => {
        const res = await request(app)
                    .post('/api/words/add')
                    .send({
                        description: "sample"
                    })
        expect(res.status).toBe(201);

    })

})