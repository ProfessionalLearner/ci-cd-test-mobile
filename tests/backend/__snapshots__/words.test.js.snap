// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test api/words/add route should get all words 1`] = `
Object {
  "__v": 0,
  "_id": "5f83322025499046105c7fc2",
  "description": Any<String>,
}
`;
